### PR TITLE
Revert "perf: hard link npm cache"

### PR DESCRIPTION
**Pull request by dsherret**
Reverts denoland/deno#22773 due to unanswered questions in https://github.com/denoland/deno/pull/22773#pullrequestreview-1923188700 (approval w/ caveats caused auto-merge) and we're doing a patch release today (unless questions can be answered before the patch).

cc @littledivy

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]

**Comment by littledivy** [comment]
Thanks for investigating and sorry about the auto-merge :sweat_smile:

**Comment by dsherret** [comment]
That's ok! I think we just need to only use auto-merge once a PR is approved and we're just waiting on the CI (or if the PR is a dep bump, typo fix, or something really simple)

**Pull request closed without merging**
